docs/source/tutorial/data-source.md

```diff
@@ -3,6 +3,424 @@ title: "2. Hook up your data sources"
 
-<h2 id="database">Connect a database</h2>
\ No newline at end of file
+To get started, install the `apollo-datasource` and `apollo-datasource-rest` packages:
+
+```bash
+npm install apollo-datasource apollo-datasource-rest --save
+```
+
+* **apollo-datasource**: This is the generic data source package. It's good for connecting to non-REST data sources.
+* **apollo-datasource-rest**: This package exposes the `RESTDataSource` class that is responsible for fetching data from a given REST API. To define a data source for the REST endpoint, extend the `RESTDataSource` class and implement the data fetching methods that your resolvers require. Let's look at a simple example to understand how data sources work.
+
+```js
+const { RESTDataSource } = require('apollo-datasource-rest');
+
+class MvrpAPI extends RESTDataSource {
+  constructor() {
+    super();
+    this.baseURL = 'https://mvrp.herokuapp.com/api/';
+  }
+
+  async getAllCars() {
+    return this.get('cars');
+  }
+
+  async getACar(plateNumber) {
+    const result = await this.get('car', {
+      plateNumber
+    });
+
+    return result[0];
+  }
+};
+
+module.exports = MvrpAPI;
+```
+
+The `https://mvrp.herokuapp.com/api/` endpoint is a simple REST API that returns data for cars. Furthermore, the `MvrpAPI` class implementation in the code above contains a `getAllCars` and `getACar` functions that wrap convenience methods provided by the `RESTDataSource` class for performing HTTP requests.
+
+In this example, the built-in `get` method used is responsible for `GET` requests. The `RESTDataSource` class provides other methods like `post`, `delete`, and `patch` for `POST`, `DELETE`, and `PATCH` requests respectively.
+
+Now that you have an understanding of how data sources work, let's hook it up for our tutorial app.
+
+Create a new `datasources` folder inside the `src` directory. This folder will contain our data source files. Now, create `launch.js` within the `datasources` directory.
+
+The REST API endpoint we'll use for our app is `https://api.spacexdata.com/v2/`. Go ahead and add the endpoint as the base URL as shown in the code below:
+
+_src/datasources/launch.js_
+
+```js
+const { RESTDataSource } = require('apollo-datasource-rest');
+
+class LaunchAPI extends RESTDataSource {
+  constructor() {
+    super();
+    this.baseURL = 'https://api.spacexdata.com/v2/';
+  }
+}
+
+module.exports = LaunchAPI;
+```
+
+In the code above, we required the `RESTDataSource` class and extended it with our custom `LaunchAPI` class. We then set up the base URL in the class constructor to the endpoint that fetches the data needed for our application.
+
+The next step is to add methods to the `LaunchAPI` class that corresponds to the type of queries our UI will fetch from the server. According to our app specifications, we'll need to get all the launches, and get a specific launch. So, let's take care of the former immediately.
+
+_src/datasources/launch.js_
+
+```js
+...
+async getAllLaunches() {
```

Comment:
Can you please update this example to the latest version? I think we're using a `launchReducer` function instead of mapping it directly.

Reply:
This section was done intentionally @peggyrayzis. The next code sample after this showed them why I did it this way and introduced them to a simple refactoring approach that birthed the `launchReducer` function.

Reply:
Done